### PR TITLE
Auto-register

### PR DESCRIPTION
**Pull request by chuck-dbos**

### CODE REVIEW

File: docs/api-reference/testing-runtime.md [hunk 1]
```diff
@@ -19,19 +19,24 @@ When writing tests, you are responsible for setting up and cleaning up your data
...
-Creates a testing runtime and loads user functions from provided `userClasses`.
+Creates a testing runtime and loads user functions from provided `userClasses`.  By default, all classes and dependencies are loaded and registered.
 Accepts an optional path to a [configuration file](./configuration.md), uses the default path (`dbos-config.yaml` in the package root) otherwise.
 
 For example, to create a runtime loading functions from the `Hello` class and using `test-config.yaml`:
 ```typescript
 testRuntime = await createTestingRuntime([Hello], "test-config.yaml");
 ```
 
+However, the defaults are generally sufficient as long as classes at least indirectly referenced from the test file:
```
Comment (qianl15) [inline docs/api-reference/testing-runtime.md:35]: Maybe move the default first (commonly used), and then explain how to create a runtime for specific classes/config file.

File: docs/api-reference/testing-runtime.md [hunk 2]
```diff
@@ -19,15 +19,20 @@ When writing tests, you are responsible for setting up and cleaning up your data
...
 
-### createTestingRuntime(userClasses, \[configFilePath\])
+### createTestingRuntime(\[userClasses\], \[configFilePath\])
 ```typescript
-async function createTestingRuntime(userClasses: object[], configFilePath: string = dbosConfigFilePath): Promise<TestingRuntime>
+async function createTestingRuntime(userClasses: object[] | undefined = undefined, configFilePath: string = dbosConfigFilePath): Promise<TestingRuntime>
 ```
 
-Creates a testing runtime and loads user functions from provided `userClasses`.
+Creates a testing runtime and loads user functions from provided `userClasses`.  By default, all classes and dependencies are loaded and registered.
```
Comment (qianl15) [inline docs/api-reference/testing-runtime.md:27]:
```suggestion
Creates a testing runtime and loads user functions from provided `userClasses`.  By default, all classes and dependencies from the test file are loaded and registered.
```

File: docs/explanations/application-structure-explanation.md [hunk 3]
```diff
@@ -34,7 +34,7 @@ The two most important files in a DBOS project are `dbos-config.yaml` and `src/o
 All options are documented in our [configuration reference](../api-reference/configuration).
 
 `src/operations.ts` is the _entrypoint_, where DBOS looks for your code.
-At startup, the DBOS runtime automatically loads all classes exported from this file, serving their endpoints and registering their decorated functions.
+At startup, the DBOS runtime automatically loads all classes exported from this file, serving their endpoints and registering their decorated functions and dependencies.
```
Comment (qianl15) [inline docs/explanations/application-structure-explanation.md:37]:
```suggestion
At startup, the DBOS runtime automatically loads all classes that are exported or (directly and indirectly) referenced from this file, serving their endpoints and registering their decorated functions.
```

File: docs/explanations/application-structure-explanation.md [hunk 4]
```diff
@@ -43,7 +43,8 @@ In a larger application, you can write your code wherever you want, but should u
 export { OperationClass1, OperationClass2 } from './FileA';
 export { OperationClass3 } from './operations/FileB';
 ```
-You can also configure a list of entrypoints using the `runtimeConfig` section of the [configuration](../api-reference/configuration#runtime).
+It is not necessary to export classes that are already referenced by the entrypoint file(s), as these will be loaded and decorated methods will be registered.
+You can configure a list of entrypoints using the `runtimeConfig` section of the [configuration](../api-reference/configuration#runtime).
```
Comment (kraftp) [inline docs/explanations/application-structure-explanation.md:47]:
```suggestion
You can also define multiple entrypoint files using the `runtimeConfig` section of the [configuration](../api-reference/configuration#runtime).
```

**Review by kraftp — APPROVED** [review]

**Review by qianl15 — APPROVED** [review]

**Pull request merged**